Week04/lesson08/homework/temp_converter/js/indexbonus3.js

```diff
@@ -0,0 +1,57 @@
+function conversion() {
+
+  var c = document.querySelector("#c").value;
+  var f = document.querySelector("#f").value;
+
+  if ((f === null || f === "") && (c === null || c === "")) {
+      document.querySelector("#alert").innerHTML = "Please fill one of the fields above.";
+  }
+
+  else if (f === null || f === "") {
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    c = c+"°C";
+  }
+
+  else if (c === null || c === "") {
+    c = (f - 32) * 5 / 9;
+    document.querySelector("#c").value = Math.round(c*10)/10+"°C";
+    f = f+"°F";
+  }
+
+  else {
+    document.querySelector("#alert").innerHTML = "Please only fill one of the fields above. Click 'Reset' to clear both fields.";
+  }
+
+}
+
+document.querySelector("#convert").onclick = conversion;
+
+
+
+
+function clear() {
+  document.querySelector("#c").value = "";
+  document.querySelector("#f").value = "";
+  document.querySelector("#alert").innerHTML = "";
+}
+
+document.querySelector("#reset").onclick = clear;
+
+
+
+
+function clearAlert() {
+      document.querySelector("#alert").innerHTML = "";
+}
+
+document.querySelector("#c").onfocus = clearAlert;
```

Comment:
To answer your first question, yes you could use just one function to clear the alert, and in fact you already have the `clear()` function (line 33 above) all set up to do just that.

You just need to do:

```
function clear() {
  document.querySelector("#c").value = "";
  document.querySelector("#f").value = "";
  document.querySelector("#alert").innerHTML = "";
}

document.querySelector("#reset").onclick = clear;
document.querySelector("#c").onfocus = clear;
document.querySelector("#f").onfocus = clear;
```

.. and get rid of both `clearAlert()` functions (by the way, the second one (line 53 below) was overriding the first one (line 44 above) so even if for some reason you wanted two separate functions, `clear()` and `clearAlert()`, you only need *one* declaration of `clearAlert()`

That said, you should know that doing it like this means that should there already be a value in the input and you click on it, it'll clear out what was there before, which depending on your desired UX may or may not be what you want.